src/common.jl

```diff
@@ -209,6 +209,113 @@ function make_dc_powerflowdata(
     )
 end
 
+function make_bus_slack_participation_factors(
+    sys::System,
+    generator_slack_participation_factors::Dict{String, Float64},
+    bus_lookup::Dict{Int, Int},
+    time_steps::Int,
+    n_buses::Int,
+    ::Matrix{PSY.ACBusTypes},
+)
+    I = Int[]
+    J = Int[]
+    V = Float64[]
+
+    for (gen_name, val) in generator_slack_participation_factors
+        val == 0.0 && continue
+        gen = PSY.get_component(PSY.Generator, sys, gen_name)
```

Comment:
You are technically allowed to have two generators with the same name as long as they don't have the same type, so this isn't guaranteed to return only one component. Maybe the keys of `generator_slack_participation_factors` should be tuples of data type, string?